joomla/joomla-cms
# Add utf8mb4 conversion for com_fields

Add missing utf8mb4 conversion for com_fields

### Summary of Changes

This pull request adds the missing utf8mb4 conversion for com_fields, which obviosly has been forgotten when adding com_fields to Joomla.

It can also serve as an example on what to do for utf8mb4 conversion when adding new tables to Joomla!

Utf8mb4 conversion is needed on mysql databases also in future, e.g. when migrating from an old database version which does not support utf8mb4 to a newer version which does.

### Testing Instructions

Can be merged by review.

Check that all tables belonging to com_fields have been caught with this PR.

Check that SQL syntax is correct for mysql.

If you don't trust that: Either perform the migration scenario mentioned above in summary of changes, or update the `#__utf8_conversion` table in phpmyadmin and set the value of column `converted` to 1, which will simulate this migration by telling the conversion that data comes from an utf8 (without mb4) database. Then go to the database fix view in the backend. It should show 1 error that the database has to be converted to utf8mb4, and using the fix button will result in all being OK without any SQL errors being shown.

### Expected result

Utf8mb4 conversion handles all Joomla core database tables.

### Actual result

Utf8mb4 conversion handles all Joomla core database tables except of those belonging to com_fields.

### Documentation Changes Required

None.
